src/gsi/setupgnssrspd.f90

```diff
@@ -0,0 +1,957 @@
+     endif
+
+     if(.not.in_curbin) cycle
+
+!    Load obs error and u,v obs
+     obserror = max(cermin(ikx),min(cermax(ikx),data(ier,i)))
+     uob = data(iuob,i)
+     vob = data(ivob,i)
+ 
+ 
+     gnssrspdob=sqrt(uob*uob+vob*vob)
+     call tintrp2a1(ges_tv,tges,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+     call tintrp2a11(ges_ps,psges,dlat,dlon,dtime,hrdifsig,&
+          mype,nfldsig)
+     call tintrp2a1(ges_lnprsl,prsltmp,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+
+     factw = data(iff10,i)
+     if(sfcmod_gfs .or. sfcmod_mm5)then
+        sfcr = data(isfcr,i)
+        skint = data(iskint,i)
+        isli=data(idomsfc,i)
+        call comp_fact10(dlat,dlon,dtime,skint,sfcr,isli,mype,factw)
+     end if
+
+     nty=ictype(ikx)
+
+     z_height = .false.
+!    if ( nty == 260 .or. nty == 261) z_height = .true.
+!    nty == 213 is temporarily assigned to SFMR retrieved wind speed from recon
+!    and is subjet to change in the future
+     if ( nty == 260 .or. nty == 261 .or. nty == 213) z_height = .true.
+
+!    Process observations reported with height differently than those
+!    reported with pressure.  Type 260=nacelle 261=tower wind spd are
+!    encoded in NCEP prepbufr files with geometric height above
+!    sea level. 
+
+!    Process cygnss observations at mslp
+     if ( nty == 298) then
+          z_height = .true.
+          data(ihgt,i) = 10
+     endif
+     
+     if (z_height) then
+        
+        drpx = zero
+        dpres = data(ihgt,i)
+        dstn = data(istnelv,i)
+        call tintrp2a11(ges_z,zsges,dlat,dlon,dtime,hrdifsig,&
+             mype,nfldsig)
+
+!       Get guess surface elevation and geopotential height profile
+!       at observation location.
+        call tintrp2a1(geop_hgtl,zges,dlat,dlon,dtime,hrdifsig,&
+             nsig,mype,nfldsig)
+
+!       Convert observation height (in dpres) from meters to grid relative
+!       units.  Save the observation height in zob for later use.
+        zob = dpres
+        call grdcrd1(dpres,zges,nsig,1)
+        factw=one
+        rlow=zero
+        rhgh=zero
+
+!       Compute observation pressure (only used for diagnostics)
+!       Set indices of model levels below (k1) and above (k2) observation.
+        if (dpres<one) then
+           z1=zero;    p1=log(psges)
+           z2=zges(1); p2=prsltmp(1)
+        elseif (dpres>nsig) then
+           z1=zges(nsig-1); p1=prsltmp(nsig-1)
+           z2=zges(nsig);   p2=prsltmp(nsig)
+           drpx = 1.e6_r_kind
+        else
+           k=dpres
+           k1=min(max(1,k),nsig)
+           k2=max(1,min(k+1,nsig))
+           z1=zges(k1); p1=prsltmp(k1)
+           z2=zges(k2); p2=prsltmp(k2)
+        endif
+
+        dz21     = z2-z1
+        dlnp21   = p2-p1
+        dz       = zob-z1
+        pobl     = p1 + (dlnp21/dz21)*dz
+        presw    = ten*exp(pobl)
+
+!    Process observations with reported pressure
+     else
```

Comment:
This setup routine processes the wind speed from GNSS, which is 10-meter wind speed. Therefore, z_height is always true. Do we need the section in which the GNSS wind speeds come with the reported pressure level (as metadata)? Could GNSS wind speed data come with pressure in the future?

Reply:
@emilyhcliu same comment as obove. Thanks.

Reply:
@emilyhcliu , may this conversation be marked as resolved?  If "yes", please do so.  If "no", what additional information do you need?